radixdlt-core/radixdlt/src/main/java/com/radixdlt/middleware2/network/ConsensusEventMessage.java

```diff
@@ -90,24 +90,32 @@
 	@DsonOutput(Output.ALL)
 	private final Vote vote;
 
-	ConsensusEventMessage() {
-		// Serializer only
-		this.proposal = null;
-		this.vote = null;
-	}
+	public ConsensusEventMessage(
+		@JsonProperty("proposal") Proposal proposal,
+		@JsonProperty("vote") Vote vote
+	) {
+		if (proposal == null && vote == null) {
+			throw new IllegalStateException("No vote nor proposal are provided for ConsensusEventMessage");
+		}
+
+		if (proposal != null && vote != null) {
+			throw new IllegalStateException("Both, vote and proposal are provided for ConsensusEventMessage");
```

Comment:
Would rather have these as `IllegalArgumentException` as it's not really an IllegalState if a node receives such as a message.

Reply:
done